models/mart/mart_speeches.sql

```diff
@@ -231,7 +227,42 @@ with
                 else null
             end as ministry_addressed
         from extract_minister_for
+    ),
+
+    agg_ministry_counts_by_topic as (
+        select
+            topic_id,
+            ministry_addressed,
+            count(*) as ministry_count,
+            rank() over (partition by topic_id order by count(*) desc) as ministry_rank
```

Comment:
read in conjunction with the suggestion for `get_majoirty_ministry`. this field is not needed.

```suggestion
```